### DISCUSSION TITLE
Get ListItem by id without loading the List first, is it possible?

### DISCUSSION THREAD

**Discussion by CodingJarod** [pnp/pnpcore#1227] [Q&A]
Hello,

I was wondering whether it is possible to load a specific list item by id without having to execute a double request.

The normal way to retrieve an item by id is the following:
```
 var list = await Context.Web.Lists.GetByTitleAsync(listTitle).ConfigureAwait(false);
 var listItem = await list.Items.GetByIdAsync(listItemId).ConfigureAwait(false);
```
but it makes a double web request.

The IList interface has a method to retrieve the list in a batch but the IListItem does not have a batch version of the GetByIdAsync method.

I've tried to do the following

```
Context.Web.Lists.GetByTitleBatchAsync(listTitle, p => p.Title,
                                                 p => p.Fields.QueryProperties(p => p.InternalName,
                                                                               p => p.FieldTypeKind,
                                                                               p => p.TypeAsString,
                                                                               p => p.Title)).ConfigureAwait(false);
 var viewXml = $$"""
                <View>
                <Query>
                    <Where>
                        <Eq>
                            <FieldRef Name='ID' />
                            <Value Type='Counter'>{{listItemId}}</Value>
                        </Eq>
                   </Where>
                </Query>
                <ViewFields>
                   <FieldRef Name='ID' />
                </ViewFields>
                <RowLimit>1</RowLimit>
                </View>
                """;
            await list.LoadItemsByCamlQueryBatchAsync(new CamlQueryOptions()
            {
                ViewXml = viewXml,
                DatesInUtc = true

            }, selectors).ConfigureAwait(false);
            await Context.ExecuteAsync().ConfigureAwait(false);
```
But on the ExecuteAsync an exception is throw with the message 'Property Id was not yet loaded' unless execute 
```
await Context.ExecuteAsync(false).ConfigureAwait(false); 
```
before loading the list item, getting back to the double web request.

In the standard CSOM model it was possible. Is there any way to do so in the SDK?

Thanks in advance.

**Comment by jansenbe** [comment]
@CodingJarod : this is not possible today, if you can cache the `IList` it can be reused to get multiple list items by id. Alternatively if you know the list id you can construct the REST/Graph request yourselves and issue a custom query, see https://pnp.github.io/pnpcore/using-the-sdk/basics-customapirequests.html for more details on how to do that.

**Reply by CodingJarod** [reply]
@jansenbe: Yes, for multiple query on the same list it will be possible to cache the list but in my use case it does not fit the common scenarios. Being the IList based on the PNPContext I'm assuming that in a custom web api scenario it's not safe to cache it across different http requests from the client. Am I correct?  
I'll definitely investigate the use of custom REST requests. Thanks for the hint.

**Reply by jansenbe** [reply]
@CodingJarod : the caching should be fine I think, when you get the IList from cache you can update it's PnPContext to the one you're currently using.
